junit-platform-ant-plugin/src/main/java/org/junit/platform/ant/plugin/JUnitPlatformPlugin.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.platform.ant.plugin;
+
+import static java.util.Optional.empty;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.tools.ant.Task;
+import org.apache.tools.ant.types.Path;
+import org.junit.platform.ant.plugin.Filters.FilterSet;
+import org.junit.platform.console.options.CommandLineOptions;
+import org.junit.platform.console.options.CommandLineOptionsParser;
+import org.junit.platform.console.options.JOptSimpleCommandLineOptionsParser;
+import org.junit.platform.console.tasks.ConsoleTask;
+import org.junit.platform.console.tasks.ConsoleTaskExecutor;
+import org.junit.platform.console.tasks.DisplayHelpTask;
+import org.junit.platform.console.tasks.ExecuteTestsTask;
+
+/**
+ * @since 1.0
+ */
+public class JUnitPlatformPlugin extends Task {
+
+	private static final String JAVA_CLASS_PATH = "java.class.path";
+
+	private static final String HELP = "help";
+	private static final String DISABLE_ANSI_COLORS = "disable-ansi-colors";
+	private static final String HIDE_DETAILS = "hide-details";
+	private static final String REPORTS_DIR = "reports-dir";
+
+	private static final String SELECT_URIS = "select-uri";
+	private static final String SELECT_FILES = "select-file";
+	private static final String SELECT_DIRECTORIES = "select-directory";
+	private static final String SELECT_PACKAGE = "select-package";
+	private static final String SELECT_CLASSES = "select-class";
+	private static final String SELECT_METHODS = "select-method";
+	private static final String SELECT_RESOURCES = "select-resource";
+
+	private static final String SCAN_CLASSPATH = "scan-classpath";
+	private static final String INCLUDE_CLASSNAMES = "include-classname";
+	private static final String INCLUDE_PACKAGE = "include-package";
+	private static final String EXCLUDE_PACKAGE = "exclude-package";
+	private static final String INCLUDE_ENGINE = "include-engine";
+	private static final String EXCLUDE_ENGINE = "exclude-engine";
+	private static final String INCLUDE_TAG = "include-tag";
+	private static final String EXCLUDE_TAG = "exclude-tag";
+
+	private final CommandLineOptionsParser commandLineOptionsParser;
+	private final ConsoleTaskExecutor consoleTaskExecutor;
+
+	private boolean printHelp = false;
+	private boolean ansiColorsDisabled = false;
+	private boolean detailsHidden = false;
+	private String reportsPath;
+	private Optional<Path> scanClasspath = empty();
+	private Optional<Filters> filters = empty();
+	private Optional<Selectors> selectors = empty();
+
+	public JUnitPlatformPlugin() {
+		super();
+		this.commandLineOptionsParser = new JOptSimpleCommandLineOptionsParser();
+		this.consoleTaskExecutor = new ConsoleTaskExecutor(System.out, System.err);
+	}
+
+	// For testing only
+	JUnitPlatformPlugin(CommandLineOptionsParser commandLineOPtionsParser, ConsoleTaskExecutor consoleTaskExecutor) {
```

Comment:
Little spelling mistake here: should read `commandLineOptionsParser` (lower p) rather than `commandLineOPtionsParser` (upper P).